lib/omise/OmiseCapabilities.php

```diff
@@ -0,0 +1,192 @@
+        if ($filters = func_get_args()) {
+            $filter   = self::combineFilters(self::argsToVariadic($filters));
+            $backends = array_filter($backends, $filter);
+            sort($backends);
+        }
+
+        return $backends;
+    }
+
+    /**
+     * Makes a filter function to check supported currency for backend.
+     *
+     * @param  string $currency
+     *
+     * @return function
+     */
+    public function makeBackendFilterCurrency($currency)
+    {
+        return function ($backend) use ($currency) {
+            return in_array(strtoupper($currency), $backend->currencies);
+        };
+    }
+
+    /**
+     * Makes a filter function to check type of backend.
+     *
+     * @param  string $type
+     *
+     * @return function
+     */
+    public function makeBackendFilterType($type)
+    {
+        return function ($backend) use ($type) {
+            return $backend->type == $type;
+        };
+    }
+
+    /**
+     * Makes a filter function to check if backends can handle given amount.
+     *
+     * @param  int $amount
+     *
+     * @return function
+     */
+    public function makeBackendFilterChargeAmount($amount)
+    {
+        $defMin = $this['limits']['charge_amount']['min'];
+        $defMax = $this['limits']['charge_amount']['max'];
+        return function ($backend) use ($amount, $defMin, $defMax) {
+            // temporary hack for now to correct min value for instalments to 500000
+            if ($backend->type == 'installment') {
+                $min = 500000;
+            } else {
+                $min = empty($backend->amount['min']) ? $defMin : $backend->amount['min'];
+            }
+            $max = empty($backend->amount['max']) ? $defMax : $backend->amount['max'];
+            return $amount >= $min && $amount <= $max;
+        };
+    }
+
+    /**
+     * Sets up 'shortcuts' to filters so they may be used thus:
+     *    $capabilities->backendFilter['currency']('THB')
+     * As well as the original:
+     *    $capabilities->makeBackendFilterCurrency('THB')
+     */
+    protected function setupFilterShortcuts()
+    {
+        foreach (self::FILTERS as $filterSubject => $availableFilters) {
+            $filterArrayName = $filterSubject . 'Filter';
+            $this->$filterArrayName = array();
+            $tempArr = &$this->$filterArrayName;
+            foreach ($availableFilters as $type) {
+                $funcName = 'make' . ucfirst($filterSubject) . 'Filter' . $type;
+                $tempArr[$type] = function () use ($funcName) {
+                    return call_user_func_array(array($this, $funcName), func_get_args());
+                };
+            }
+        }
+    }
+
+    /**
+     * Combines boolean filters.
+     *
+     * @param  [functions] $filters
+     *
+     * @return function
+     */
+    private static function combineFilters($filters)
+    {
+        return function ($value) use ($filters) {
+            foreach ($filters as $filter) {
```

Comment:
Code diarrhoea. The code equivalent of div-itis in HTML

Reply:
PHP Code Sniffer reason (same as the one I commented above).